### PR TITLE
Introduce option to add all parts to BOM for assembly

### PR DESCRIPTION
**Pull request by sl0**
Using this option adds all unspecified components to the BOM without specifying a part number so JPCPCB.com can choose on their guesses... Just to empower the engineers laziness ;-)

### CODE REVIEW

File: jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import csv
 import re
 import sys
-from jlc_kicad_tools.logger import Log
+import logging
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py:21]: Why did you remove our custom logging?

File: jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py [hunk 2]
```diff
@@ -59,46 +55,22 @@ def FixRotations(input_filename, output_filename, db):
...
-        if posx_index is None:
-          _LOGGER.logger.warning("Failed to find 'PosX' column in the csv file")
+          logging.warning("Failed to find 'Rot' column in the csv file")
           return False
         # Replace column names with labels JLC wants.
         for i in range(len(row)):
           if row[i] in HEADER_REPLACEMENT_TABLE:
             row[i] = HEADER_REPLACEMENT_TABLE[row[i]]
       else:
-        rotation = float(row[rotation_index])
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py:83]: I assume this change and the one below are unintentional?

File: jlc_kicad_tools/jlc_lib/kicad_netlist_reader.py [hunk 3]
```diff
@@ -262,6 +260,7 @@ def get(self, elemName, attribute="", attrmatch=""):
             ret = child.get(elemName, attribute, attrmatch)
             if ret != "":
                 if type(ret) != str: ret = ret.encode('utf-8')
+                # print("R: ", ret)
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/kicad_netlist_reader.py:263]: Please remove commented out code

File: jlc_kicad_tools/jlc_lib/kicad_netlist_reader.py [hunk 4]
```diff
@@ -446,6 +446,7 @@ def __init__(self, fname=""):
         fname -- The name of the generic netlist file to open (Optional)
 
         """
+        print('reading: ', fname)
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/kicad_netlist_reader.py:449]: Stray print?

File: jlc_kicad_tools/jlc_lib/generate_bom.py [hunk 5]
```diff
@@ -69,33 +67,32 @@ def GenerateBOM(input_filename, output_filename, opts):
...
     # Check part numbers for uniqueness
-    if len(lcsc_part_numbers) == 0 and not opts.include_all_groups:
+    if len(lcsc_part_numbers) == 0:
       if opts.warn_no_partnumber:
-        _LOGGER.logger.warning("No LCSC part number found for components {}".format(",".join(refs)))
-      continue
-    elif len(lcsc_part_numbers) > 1:
-      _LOGGER.logger.error("Components {components} from same group have different LCSC part numbers: {partnumbers}".format(
+        logging.warning("No LCSC part number found for components {}".format(",".join(refs)))
+      if not opts.all_parts_lcsc:
```
Comment (matthewlai) [inline jlc_kicad_tools/jlc_lib/generate_bom.py:73]: Where is this defined?

**Review by matthewlai — CHANGES_REQUESTED** [review]
Also, don't we already support this with the '--include-all-component-groups' option?

File: jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py (hunk 1 above)
Comment (sl0) [reply jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py:21]: Sorry, i did not see an option like that. My start was at your commit 16c30db8c1a07f47bacc4fccce14b1db3205a048
I also did not delete anything exept the tabs, which i replace by more pythonic spaces.

File: jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py (hunk 2 above)
Comment (sl0) [reply jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py:83]: Sorry, I did not change that file. git log shows e2e447e73027acd21f1bdbec3e447fc0506f218b as last change of that file,

File: jlc_kicad_tools/jlc_lib/generate_bom.py (hunk 5 above)
Comment (sl0) [reply jlc_kicad_tools/jlc_lib/generate_bom.py:73]: I'm very sorry about all my confusing changes. I beg your pardon.
I'm going to do it onec again from startup. And before desturbing you again, I will try and test all the present options.
Thank you very much for your reading,
kind reagards from Cologne, Germany
yours Johannes

Comment (matthewlai) [reply jlc_kicad_tools/jlc_lib/generate_bom.py:73]: No worries. It looks like you are working from an old version. The "--include-all-component-groups" version was added somewhat recently, and if you were working from an old commit you have to rebase it to the current HEAD before sending a pull request, because otherwise your changes will also revert all the changes between your starting commit and HEAD.

**Comment by sl0** [comment]
Sorry, there are too mmany mistakes, I made. I close this pull_request, and I'll investigate further more before I create a new pull-request.
Thank you very much, kind regards
Johannes
